lib/libzpool/zrepl_mgmt.c

```diff
@@ -607,3 +607,49 @@ uzfs_zinfo_destroy_internal_clone(zvol_info_t *zinfo)
 	    clone_zv);
 	return (ret);
 }
+
+/*
+ * This API is used to delete stale
+ * cloned volume and backing snapshot.
+ */
+int
+uzfs_zinfo_destroy_stale_clone(zvol_info_t *zinfo)
+{
+	int ret = 0;
+	char *clone_subname = NULL;
+	zvol_state_t *l_snap_zv = NULL, *l_clone_zv = NULL;
+	zvol_state_t *zv;
+
+	if (!zinfo->main_zv)
+		return (0);
+
+	zv = zinfo->main_zv;
+
+	ret = get_snapshot_zv(zv, REBUILD_SNAPSHOT_SNAPNAME,
+	    &l_snap_zv, B_FALSE, B_TRUE);
+	if (ret != 0) {
+		LOG_ERR("Failed to get info about %s@%s",
+		    zv->zv_name, REBUILD_SNAPSHOT_SNAPNAME);
+		return (ret);
+	}
+
+	clone_subname = kmem_asprintf("%s_%s", strchr(zv->zv_name, '/') + 1,
+	    REBUILD_SNAPSHOT_CLONENAME);
+
+	ret = uzfs_open_dataset(zv->zv_spa, clone_subname, &l_clone_zv);
+	if (ret == 0) {
```

Comment:
we should handle error here. Any reason why we are not handling error?

Reply:
No. I will update this.